Cargo.toml

```diff
@@ -6,25 +6,48 @@ edition = "2021"
 license = "MIT OR Apache-2.0"
 
 [dependencies]
-hal = { package = "esp32c3-hal", version = "0.15.0", features = ["embassy", "async", "embassy-executor-thread", "embassy-time-timg0"] }
-esp-wifi = { version = "0.3.0", features = ["esp32c3", "async", "embassy-net", "wifi", "ipv6"] }
-log = "0.4.16"
-nb = "1.0.0"
+esp-hal = { version = "0.22.0", features = [
+    "esp32c3",
+] }
+esp-wifi = { version = "0.11.0", default-features = false, features = [
+    "esp32c3",
+    "utils",
+    "wifi",
+    "esp-alloc",
+    "log",
+] }
+esp-alloc = { version = "0.5.0" }
+log = "0.4.22"
```

Comment:
Are you using log? Briefly looking, it's not used, it might be a good idea to switch from `println` to logs.